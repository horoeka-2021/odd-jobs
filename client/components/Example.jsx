import React, { useEffect, useState } from 'react'

// import { getExamples } from '../api/example'

function Example (props) {
  const [userInput, setUserInput] = useState('')
  const [contentFromDB, setContentFromDB] = useState({})

  useEffect(() => {
    fetchExamples()
  })

  function fetchExamples () {
    const obj = {
      title: 'Test title'
    }
    setContentFromDB(obj)
  }

  function handleClick (event) {
    setUserInput(userInput.toUpperCase())
  }

  function handleChange (event) {
    setUserInput(event.target.value)
  }

  return (
    <div>
      <h2>Example React component - props.title:{props.title}</h2>
      <p>The text you entered:{userInput}</p>
      <input onChange={handleChange} value={userInput}/>
      <button onClick={handleClick}>Click</button>
      <p>This is from database:{contentFromDB.title}</p>
    </div>
  )
}

export default Example
