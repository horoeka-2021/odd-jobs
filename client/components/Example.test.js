/**
 * @jest-environment jsdom
 */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { expect } from '@jest/globals'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

import renderer from 'react-test-renderer'

import Example from './Example'
require('regenerator-runtime/runtime')

// Structural test using snapshots
test('Example - client -  snapshot testing using testing-library', () => {
  const { asFragment } = render(<Example title='changed title' />)

  expect(asFragment()).toMatchSnapshot()
})

test('Example - client - snapshot testing using react-test-renderer', () => {
  const app = renderer.create(<Example />)

  expect(app.toJSON()).toMatchSnapshot()
})

test('Example - client - UI testing', () => {
  render(<Example title='UI testing' />)
  const paragraph = screen.getByText('The text you entered', { exact: false })
  expect(paragraph.textContent).toMatch('')

  const textbox = screen.getByRole('textbox')
  userEvent.type(textbox, 'abcdef')

  const button = screen.getByRole('button')
  userEvent.click(button)

  expect(paragraph.textContent).toMatch('ABCDEF')
  expect(paragraph.textContent).not.toMatch('abcdefq')
})
