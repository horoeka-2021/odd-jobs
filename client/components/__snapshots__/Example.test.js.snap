// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example - snapshot testing using react-test-renderer 1`] = `
<div>
  <h2>
    Example React component - props.title:
  </h2>
  <p>
    The text you entered:
    
  </p>
  <input
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Click
  </button>
</div>
`;

exports[`Example - snapshot testing using testing-library 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Example React component - props.title:changed title
    </h2>
    <p>
      The text you entered:
    </p>
    <input
      value=""
    />
    <button>
      Click
    </button>
  </div>
</DocumentFragment>
`;
